app/forms/state_file/taxes_owed_form.rb

```diff
@@ -39,5 +41,30 @@ def date_electronic_withdrawal
     def date_electronic_withdrawal_is_valid_date
       valid_text_date(date_electronic_withdrawal_year, date_electronic_withdrawal_month, date_electronic_withdrawal_day, :date_electronic_withdrawal)
     end
+
+    def withdraw_amount_higher_than_owed?
+      owed_amount = intake.calculated_refund_or_owed_amount.abs
+      if self.withdraw_amount.to_i > owed_amount
+        self.errors.add(
+          :withdraw_amount,
+          "Please enter in an amount less than or equal to #{owed_amount}"
+        )
+      end
+    end
+
+    def withdraw_amount_lower_than_owed?
+      owed_amount = intake.calculated_refund_or_owed_amount.abs
```

Comment:
So this will not allow them to pay partially on our app and partially online?

Reply:
It will, the validation on the taxes owed form will allow them to pay a portion and then the rest with the link given.

https://user-images.githubusercontent.com/2385700/283914631-6be12c07-d167-4712-bdaf-e9f1ad6dec8b.png

Reply:
Updated